matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/sync/job/SyncWorker.kt

```diff
@@ -47,50 +45,45 @@ internal class SyncWorker(context: Context,
             val automaticallyRetry: Boolean = false
     )
 
-    @Inject lateinit var syncAPI: SyncAPI
-    @Inject lateinit var filterRepository: FilterRepository
-    @Inject lateinit var syncResponseHandler: SyncResponseHandler
-    @Inject lateinit var sessionParamsStore: SessionParamsStore
-    @Inject lateinit var syncTokenStore: SyncTokenStore
-
+    @Inject
+    lateinit var syncTask: SyncTask
+    @Inject
+    lateinit var taskExecutor: TaskExecutor
 
     override suspend fun doWork(): Result {
         Timber.i("Sync work starting")
         val params = WorkerParamsFactory.fromData<Params>(inputData) ?: return Result.success()
         val sessionComponent = getSessionComponent(params.userId) ?: return Result.success()
         sessionComponent.inject(this)
 
-        val requestParams = HashMap<String, String>()
-        requestParams["timeout"] = params.timeout.toString()
-        requestParams["filter"] = filterRepository.getFilter()
-        val token = syncTokenStore.getLastToken()?.also { requestParams["since"] = it }
-        Timber.i("Sync work last token $token")
-
-        return executeRequest<SyncResponse> {
-            apiCall = syncAPI.sync(requestParams)
-        }.fold(
-                {
-                    if (it is Failure.ServerError
-                            && it.error.code == MatrixError.UNKNOWN_TOKEN) {
-                        sessionParamsStore.delete(params.userId)
-                        Result.failure()
-                    } else {
-                        Timber.i("Sync work failed $it")
-                        Result.retry()
+
+        val latch = CountDownLatch(1)
```

Comment:
Maybe using runBlocking would be cleaner

Reply:
ok, P2 :)